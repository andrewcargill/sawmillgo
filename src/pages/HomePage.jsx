import React from 'react';
import Typography from '@mui/material/Typography'
import Grid from '@mui/material/Grid'
import { List, ListItem, ListItemAvatar, ListItemText, Paper } from '@mui/material';
import ParkIcon from '@mui/icons-material/Park';
import QrCodeIcon from '@mui/icons-material/QrCode';
import HandymanIcon from '@mui/icons-material/Handyman';
import VerifiedIcon from '@mui/icons-material/Verified';
import BarChartIcon from '@mui/icons-material/BarChart';

const HomePage = () => {
 

  return (
    <>
    <Grid container pb={4}>
     <Typography variant="h2" color="initial">A cloud based forestry management system that delivers complete transparency and the story behind your wood products.</Typography>
    </Grid>

    <Grid container p={3} bgcolor={'primary.main'} color={'primary.contrastText'}  justifyContent={'center'}>
      <Paper>
      <List>
        <ListItem>
          <ListItemAvatar>
            <ParkIcon />
            </ListItemAvatar>
            <ListItemText 
            primary="Source Transparency" 
            secondary=" Each tree is logged with details like the lumberjack's name, removal reason, exact GPS location, and a photo." />
          </ListItem>
  
        <ListItem>
          <ListItemAvatar>
            <BarChartIcon />
            </ListItemAvatar>
            <ListItemText 
            primary="Processing Insight" 
            secondary=" From logging to milling, every step is timestamped with operator details and visuals from the sawmill." />
          </ListItem>
        <ListItem>
          <ListItemAvatar>
            <VerifiedIcon />
            </ListItemAvatar>
            <ListItemText 
            primary="Quality Assurance" 
            secondary=" Follow the wood's drying process through comprehensive data and drying history graphs." />
          </ListItem>
        <ListItem>
          <ListItemAvatar>
            <HandymanIcon />
            </ListItemAvatar>
            <ListItemText 
            primary="Craftsmanship Tracking" 
            secondary=" Artisans document their creative process, turning planks into unique products." />
          </ListItem>
        <ListItem>
          <ListItemAvatar>
            <QrCodeIcon />
            </ListItemAvatar>
            <ListItemText 
            primary="Final Product Traceability" 
            secondary="Every product comes with a QR code that links to a detailed report about the woodâ€™s journey and origins." />
          </ListItem>
        </List>
        </Paper>
      
    </Grid>

    {/* <Grid container justifyContent={'center'}>
     <Typography variant="h2" color="initial">Maximize the Value of Your Wood.</Typography>
    </Grid> */}

    </>
  );
};

export default HomePage;
