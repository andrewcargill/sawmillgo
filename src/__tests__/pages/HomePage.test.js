import React from "react";
import { render, screen, fireEvent } from "@testing-library/react";
import { MemoryRouter } from "react-router-dom"; // For routing tests
import HomePage from "../../pages/HomePage";

describe("HomePage Tests", () => {
  test("renders homepage elements correctly", () => {
    render(
      <MemoryRouter>
        <HomePage />
      </MemoryRouter>
    );

    // Check for headings, buttons, and images
    expect(
      screen.getByRole("heading", { name: /welcome to sawmillgo/i })
    ).toBeInTheDocument();
    expect(screen.getByRole("link", { name: /discover more/i })).toBeInTheDocument();
    expect(screen.getByAltText("SawmillGo Overview")).toBeInTheDocument();
  });

  

  test("renders the correct introductory text", () => {
    render(
      <MemoryRouter>
        <HomePage />
      </MemoryRouter>
    );

    expect(
      screen.getByText(/data-driven forestry for a sustainable future/i)
    ).toBeInTheDocument();
    expect(
      screen.getByText(/empowers forest owners, sawmills, and creators/i)
    ).toBeInTheDocument();
  });

  test("renders hero image with correct alt text and source", () => {
    render(
      <MemoryRouter>
        <HomePage />
      </MemoryRouter>
    );

    const heroImage = screen.getByAltText("SawmillGo Overview");
    expect(heroImage).toHaveAttribute("src", "home_hero2.webp");
  });

  test("matches snapshot", () => {
    const { asFragment } = render(
      <MemoryRouter>
        <HomePage />
      </MemoryRouter>
    );

    expect(asFragment()).toMatchSnapshot();
  });
});
