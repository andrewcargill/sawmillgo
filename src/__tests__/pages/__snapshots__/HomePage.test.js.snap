// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomePage Tests matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="MuiContainer-root MuiContainer-maxWidthLg css-1oqqzyl-MuiContainer-root"
    style="padding: 20px;"
  >
    <div
      class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-4 css-xtlx1y-MuiGrid-root"
      style="margin-bottom: 40px;"
    >
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-6 css-1fyyp8j-MuiGrid-root"
      >
        <h1
          class="MuiTypography-root MuiTypography-h2 MuiTypography-gutterBottom fade-in-2 css-i1hrn3-MuiTypography-root"
        >
          Welcome to SawmillGo
        </h1>
        <p
          class="MuiTypography-root MuiTypography-h6 MuiTypography-paragraph fade-in-5 css-1mo68eo-MuiTypography-root"
        >
          Data-driven forestry for a sustainable future.
        </p>
        <p
          class="MuiTypography-root MuiTypography-body1 MuiTypography-paragraph fade-in-3 css-z2eky3-MuiTypography-root"
        >
          SawmillGo empowers forest owners, sawmills, and creators to enhance transparency and authenticity through verified data. Our system supports sustainable practices, giving stakeholders the tools they need to increase both environmental impact and profitability.
        </p>
        <a
          class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeLarge MuiButton-containedSizeLarge MuiButton-colorPrimary MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeLarge MuiButton-containedSizeLarge MuiButton-colorPrimary fade-in-4 css-jh47zj-MuiButtonBase-root-MuiButton-root"
          href="/concept"
          tabindex="0"
        >
          discover more
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </a>
      </div>
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-6 css-1fyyp8j-MuiGrid-root"
      >
        <img
          alt="SawmillGo Overview"
          class="MuiCardMedia-root MuiCardMedia-media MuiCardMedia-img fade-in-1 css-o69gx8-MuiCardMedia-root"
          src="home_hero2.webp"
        />
      </div>
    </div>
  </div>
</DocumentFragment>
`;
